src/fr/anisama/src/eu/kanade/tachiyomi/animeextension/fr/anisama/extractors/VidCdnExtractor.kt

```diff
@@ -0,0 +1,53 @@
+package eu.kanade.tachiyomi.animeextension.fr.anisama.extractors
+
+import eu.kanade.tachiyomi.animesource.model.Video
+import eu.kanade.tachiyomi.network.GET
+import eu.kanade.tachiyomi.util.parseAs
+import kotlinx.serialization.Serializable
+import okhttp3.Headers
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.OkHttpClient
+import okhttp3.internal.commonEmptyHeaders
+
+class VidCdnExtractor(
+    private val client: OkHttpClient,
+    headers: Headers = commonEmptyHeaders,
+) {
+
+    private val headers = headers.newBuilder()
+        .set("Referer", "https://msb.toonanime.xyz")
+        .build()
+
+    @Serializable
+    data class CdnSourceDto(val file: String)
+
+    @Serializable
+    data class CdnResponseDto(val sources: List<CdnSourceDto>)
+
+    fun videosFromUrl(
+        url: String,
+        videoNameGen: (String) -> String = { quality -> quality },
+    ): List<Video> {
+        return runCatching {
```

Comment:
Remove runcatching from the extractor, you're catching such errors in the main class anyways